JavaScripts/forEachMapReduceFilter.js

```diff
@@ -0,0 +1,205 @@
+  "fruit-by-the-foot",
+  "pogs",
+  "my room",
+  "temporary tattoos",
+];
+const myStuff = [
+  "rules",
+  "fruit-by-the-foot",
+  "wedgies",
+  "sweaters",
+  "skateboards",
+  "family-night",
+  "my room",
+  "braces",
+  "the information superhighway",
+];
+function justCoolStuff(array1, array2) {
+  return array1.filter((element, index, array) => {
+    return array2.includes(element);
+  });
+}
+console.log(justCoolStuff(myStuff, coolStuff));
+// Should print [ 'fruit-by-the-foot', 'skateboards', 'my room' ]
+
+///////////////////
+// every() - returns boolean if condition for every element is fulfilled
+function isTheDinnerVegan(arrayOfObjects){
+  return arrayOfObjects.every(
+    (element, index, array) => {
+      return element.source === 'plant';
+    }
+  )
+}
+const dinner = [{name: 'hamburger', source: 'meat'}, {name: 'cheese', source: 'dairy'}, {name: 'ketchup', source:'plant'}, {name: 'bun', source: 'plant'}, {name: 'dessert twinkies', source:'unknown'}];
+console.log(isTheDinnerVegan(dinner))
+// Should print false
+
+
+///////////////////
+// reduce() - additional parameter 'sumThusFar' (see sumedUp function)
+
+// 1
+let totaled = singleArray.reduce(sumedUp, 10); // 10 is a 2nd parameter to add to the total sum (acts as a starting point for the counting up)
+console.log(totaled);
+// 2
+let totaled2 = multiArray.map((element) => element.reduce(sumedUp));
+console.log(totaled2);
+// 3 reduce() works on strings too!
+let animalsReduced = animals.reduce((accumulator, currentElement) => {
+  return accumulator + currentElement[0];
+}, "");
+console.log(animalsReduced);
+
+///////////////////
+// sort()
+
+// 1
+const years = [1970, 1999, 1951, 1982, 1963, 2011, 2018, 1922];
+// no function passed in will sort ascending
+console.log(years.sort());
+// with function passed in can be used to sort descending
+function sortYears(arrayOfYears) {
+  return arrayOfYears.sort((a, b) => {
+    return a < b;
+  });
+}
+console.log(sortYears(years));
+
+// 2 sorting array of objects by their attributes
+const speciesArray = [ {speciesName:'shark', numTeeth:50}, {speciesName:'dog', numTeeth:42}, {speciesName:'alligator', numTeeth:80}, {speciesName:'human', numTeeth:32}];
+function sortSpeciesByTeeth(arrayOfObject){
+  return arrayOfObject.sort(
+    (a,b) => {return a.numTeeth > b.numTeeth}
+  )
+}
+console.log(sortSpeciesByTeeth(speciesArray))
+// Should print:
+// [ { speciesName: 'human', numTeeth: 32 },
+//   { speciesName: 'dog', numTeeth: 42 },
+//   { speciesName: 'shark', numTeeth: 50 },
+//   { speciesName: 'alligator', numTeeth: 80 } ]
+
+///////////////////
+// .findIndex() and .indexOf()
+
+// using .indexOf()
+function findMyKeys2(arrayOfStrings) {
+  if (arrayOfStrings.includes("keys")) {
+    return arrayOfStrings.indexOf("keys");
+  } else {
+    return -1;
+  }
+}
+// using .findIndex()
+function findMyKeys(arrayOfStrings) {
+  return arrayOfStrings.findIndex(
+    (element) => {return element === "keys";})
```

Comment:
Nitpick: one liners with arrow functions can omit the braces and return keyword and trailing semicolon:

(element) => element === "keys"